### PR TITLE
Solve labels-images association in embedding

### PR DESCRIPTION
**Pull request by lucabergamini**
TB needs a squared sprite, even if the number of images is not a perfect square, so we need to ensure to always present a squared grid. If the number of images is not a perfect square (i.e 8 is not), the easiest way is to cat images (random noise will do the trick) to reach the nearest big square ( 9 in this case).

### CODE REVIEW

File: tensorboard/embedding.py [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
     import torch
     import torchvision
-    nrow = int(math.floor(math.sqrt(label_img.size(0))))
-    xx = torchvision.utils.make_grid(torch.Tensor(1,3,32,32), padding=0)
-    if xx.size(2)==33: # https://github.com/pytorch/vision/issues/206
+    # this ensure we have enought space for the images
+    nrow = int(math.ceil((label_img.size(0)) ** 0.5))
+    # cat the images to reach the square of base_size
+    label_img = torch.cat((label_img, torch.randn(nrow ** 2 - label_img.size(0), *label_img.size()[1:]) * 255), 0)
+    # this ensure no pixel are appended by make_grid call in save_image (such a stupid function)
```
Comment (lanpa) [inline tensorboard/embedding.py]: You means line 20?

Comment (lucabergamini) [inline tensorboard/embedding.py]: yep :) I suggest random instead of zeros, because if something goes wrong in the visualization, noise is more easily recognizable.

Comment (lanpa) [inline tensorboard/embedding.py]: My question is why there will be no additional pixel in `make_grid()` call if you send a square number of images?

Comment (lucabergamini) [inline tensorboard/embedding.py]: No that's relies on your code (lines 21 to end)

File: tensorboard/embedding.py [hunk 2]
```diff
@@ -91,12 +93,12 @@ def add_embedding(mat, save_path, metadata=None, label_img=None):
     except OSError:
         print('warning: dir exists')
     if metadata is not None:
-        assert mat.size(0)==len(metadata), '#labels should equal with #data points'
+        assert mat.size(0) == len(metadata), '#labels should equal with #data points'
```
Comment (lanpa) [inline tensorboard/embedding.py]: other functions in this project do not have space around comparison.

File: tensorboard/embedding.py (hunk 1 above)
Comment (lucabergamini) [inline tensorboard/embedding.py]: I put comments above btw

File: tensorboard/embedding.py (hunk 2 above)
Comment (lucabergamini) [inline tensorboard/embedding.py]: Yeah it's Pycharm auto-fixes..

Comment (lanpa) [inline tensorboard/embedding.py]: Would you mind I revert it?

Comment (lucabergamini) [inline tensorboard/embedding.py]: absolutely no problem :)

**Comment by lanpa** [comment]
fixed #12
